### PR TITLE
Add python tests for Thermostat presets feature

### LINKED ISSUE #34640: Add support for automated test scripts (python) for testing Thermostat Preset features
Right now we have manual tests for testing Presets feature in the thermostat cluster. It would be preferable to have automated test scripts for this.

### PR DESCRIPTION
**Pull request by nivi-apple**
Fixes: #34640

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
Fast tracking tests

File: src/python_testing/TC_TSTAT_4_2.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs: run1
+# test-runner-run/run1/app: ${ALL_CLUSTERS_APP}
```
Comment (tcarmelveilleux) [inline src/python_testing/TC_TSTAT_4_2.py:22]: Does this need to be the thermostat app?

File: src/python_testing/TC_TSTAT_4_2.py [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+            status_ok = (status == Status.Success)
+            asserts.assert_true(status_ok, "Presets write did not return Success as expected")
+
+            # Send the CancelPresetsSchedulesRequest command
+            await self.send_cancel_preset_request_command()
+
+            # Send the CommitPresetsSchedulesRequest command and expect InvalidInState as the previous edit request was cancelled
+            await self.send_commit_preset_request_command(expected_status=Status.InvalidInState)
+
+        # TODO: Add tests for the total number of Presets exceeds the NumberOfPresets supported. Also Add tests for adding presets with preset scenario not present in PresetTypes.
```
Comment (tcarmelveilleux) [inline src/python_testing/TC_TSTAT_4_2.py:344]: Please create a follow-up issue

**Review by tcarmelveilleux — APPROVED** [review]

File: src/python_testing/TC_TSTAT_4_2.py (hunk 1 above)
Comment (nivi-apple) [reply src/python_testing/TC_TSTAT_4_2.py:22]: i added support to all-clusters-app for Presets attributes so its good

File: src/python_testing/TC_TSTAT_4_2.py (hunk 2 above)
Comment (nivi-apple) [reply src/python_testing/TC_TSTAT_4_2.py:344]: Filed #https://github.com/project-chip/connectedhomeip/issues/34695

**Pull request merged**